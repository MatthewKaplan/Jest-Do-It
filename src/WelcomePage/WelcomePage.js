import React from 'react';
import '../Styles/_WelcomePage.scss';

const WelcomePage = (props) => {
  return (
    <section className="welcomePage">
      <article className="backGround">
        <div className="welcomeContent">
          <span className="welcomeHeader">Welcome to Jest Do It</span>
          <h2 className="subHeader">An online resource for learning the testing framework Jest along with the testing utility Enzyme.</h2>
          <img src="https://i.imgur.com/G1SZXAY.png" alt="Jest Icon" className="welcomeImg"/><br/>
          <form className="signInForm">
            <input
              type='text'
              placeholder='Enter your name to continue'
              value={props.playerName}
              onChange={props.setPlayer}
              className='nameInput'
              data-test='user-name-input'
              maxLength="15" 
            />
            <input className="startGame" 
              type='submit'
              disabled={!props.playerName}
              data-test='btn-start-quiz'
              onClick={props.startQuiz}
              value='Start Game'
            />
          </form>
        </div>
      </article>
    </section>
  )
}

export default WelcomePage;