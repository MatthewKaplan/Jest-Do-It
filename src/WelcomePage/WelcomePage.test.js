import React from 'react';
import { WelcomePage } from './WelcomePage';
import { shallow } from 'enzyme';
