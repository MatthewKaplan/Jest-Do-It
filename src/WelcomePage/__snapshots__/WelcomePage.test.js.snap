// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomePage should match snapshot 1`] = `
<section
  className="welcomePage"
>
  <article
    className="backGround"
  >
    <div
      className="welcomeContent"
    >
      <span
        className="welcomeHeader"
      >
        Welcome to Jest Do It
      </span>
      <h2
        className="subHeader"
      >
        An online resource for learning the testing framework Jest along with the testing utility Enzyme.
      </h2>
      <img
        alt="Jest Icon"
        className="welcomeImg"
        src="https://i.imgur.com/G1SZXAY.png"
      />
      <br />
      <form
        className="signInForm"
        data-test="start-quiz-btn"
      >
        <input
          className="nameInput"
          data-test="user-name-input"
          maxLength="15"
          placeholder="Enter your name to continue"
          type="text"
        />
        <input
          className="startGame"
          data-test="btn-start-quiz"
          type="submit"
          value="Start Game"
        />
      </form>
    </div>
  </article>
</section>
`;
